src/protagonist/API/Features/NamedQueries/Requests/DeleteNamedQuery.cs

```diff
@@ -0,0 +1,50 @@
+﻿using API.Infrastructure.Requests;
+using DLCS.Core;
+using DLCS.Model.Messaging;
+using DLCS.Repository;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.Features.NamedQueries.Requests;
+
+public class DeleteNamedQuery: IRequest<ResultMessage<DeleteResult>>
+{
+    public DeleteNamedQuery(int customerId, string namedQueryId)
+    {
+        CustomerId = customerId;
+        NamedQueryId = namedQueryId;
+    }
+    
+    public int CustomerId { get; }
+    
+    public string NamedQueryId { get; }
+}
+
+public class DeleteNamedQueryHandler : IRequestHandler<DeleteNamedQuery, ResultMessage<DeleteResult>>
+{
+    private readonly DlcsContext dbContext;
+
+    public DeleteNamedQueryHandler(DlcsContext dbContext)
+    {
+        this.dbContext = dbContext;
+    }
+
+    public async Task<ResultMessage<DeleteResult>> Handle(DeleteNamedQuery request, CancellationToken cancellationToken)
+    {
+        var deleteResult = DeleteResult.NotFound;
+        var message = string.Empty;
+        
+        var namedQuery = await dbContext.NamedQueries.AsNoTracking().SingleOrDefaultAsync(
```

Comment:
`AsNoTracking()` feels odd here as we are then deleting it. The call to `dbContext.NamedQueries.Remove()` would being change tracking anyway so I think this should go.

Reply:
Good point - fixed in cac92e7f7f58f765c9229630a3373e9ae1a6f1ac